agents/ibootbar/ibootbar.py

```diff
@@ -0,0 +1,411 @@
+            oid_cache[field_name]["name"] = names[int(field_name[-1])]
+            oid_cache[field_name]["description"] = oid_description
+            oid_cache['ibootbar_connection'] = {'last_attempt': time.time(),
+                                                'connected': True}
+            oid_cache['timestamp'] = timestamp
+    # This is a TypeError due to nothing coming back from the yield,
+    # so get_result is None here and can't be iterated.
+    except TypeError:
+        oid_cache['ibootbar_connection'] = {'last_attempt': time.time(),
+                                            'connected': False}
+        raise ConnectionError('No SNMP response. Check your connection.')
+
+    return oid_cache
+
+
+class ibootbarAgent:
+    """Monitor the ibootbar system via SNMP.
+
+    Parameters
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    address : str
+        Address of the ibootbar.
+    port : int
+        SNMP port to issue GETs to, default to 161.
+    version : int
+        SNMP version for communication (1, 2, or 3), defaults to 2.
+
+    Attributes
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    is_streaming : bool
+        Tracks whether or not the agent is actively issuing SNMP GET commands
+        to the ibootbar. Setting to false stops sending commands.
+    log : txaio.tx.Logger
+        txaio logger object, created by the OCSAgent
+    """
+
+    def __init__(self, agent, address, port=161, version=2):
+        self.agent = agent
+        self.is_streaming = False
+        self.log = self.agent.log
+        self.lock = TimeoutLock()
+
+        self.log.info(f'Using SNMP version {version}.')
+        self.version = version
+        self.snmp = SNMPTwister(address, port)
+
+        self.lastGet = 0
+
+        agg_params = {
+            'frame_length': 10 * 60  # [sec]
+        }
+        self.agent.register_feed('ibootbar',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=0)
+
+    @inlineCallbacks
+    def acq(self, session, params=None):
+        """acq()
+
+        **Process** - Fetch values from the ibootbar via SNMP.
+
+        Notes:
+            The most recent data collected is stored in session.data in the
+            structure::
+                >>> response.session['data']
+                {"fields":
+                    {ibootbar:
+                        {outletStatus_0: {"status": 1, "name": Outlet-1, "description": "on"},
+                         outletStatus_1: {"status": 0, "name": Outlet-2, "description": "off"},
+                         ...
+                         ibootbar_connection: {'last_attempt': 1656085022.680916, 'connected': True},
+                         timestamp: {1656085022.680916}
+                        }
+                    }
+                }
```

Comment:
Based on `update_cache()` I think the structure here might be flatter than this. It looks like the "field" and "ibootbar" levels are not there, i.e. it should read:
```
>>> response.session['data']
{outletStatus_0: {"status": 1, "name": Outlet-1, "description": "on"},
 outletStatus_1: {"status": 0, "name": Outlet-2, "description": "off"},
 ...
 ibootbar_connection: {'last_attempt': 1656085022.680916, 'connected': True},
 timestamp: {1656085022.680916}}
```

You can confirm this structure by running the Agent and using a client to run:
```
>>> response = client.acq.status()
>>> response.session['data']
```